File: frontend/src/types/Note.ts
import Tag from "./Tag"

export default interface Note {
    id?: any | null,
    header: string,
    body: string,
    tags: Tag[]
    // color: string,
    // edited: string,
}

export type NoteRenderData = {
    id: number
    header: string
    tags: Tag[]
}

File: backend/internal/model/dto/note.go
package dto

import (
	"neatly/internal/model"
)

type CreateNoteDTO struct {
	Header string `json:"header" binding:"required"`
	Color  string `json:"color" binding:"required"`
	Body   string `json:"body"`
}

type UpdateNoteDTO struct {
	ID     int
	Header string `json:"header"`
	Body   string `json:"body"`
	Color  string `json:"color"`
}

type GetAllNotesDTO struct {
	Notes []model.Note `json:"notes"`
}
